experimental/LieAlgebras/test/iso_gap_oscar-test.jl

```diff
@@ -0,0 +1,110 @@
+          ),
+        ),
+      ]
+
+      @testset for LG in lie_algebras
+        iso = Oscar.iso_gap_oscar(LG)
+        @test domain(iso) == LG
+        LO = codomain(iso)
+        @test LO isa AbstractLieAlgebra
+        @test codomain(Oscar.iso_oscar_gap(LO)) === LG
+
+        @test iso == Oscar.iso_gap_oscar(LG) # test caching
+
+        for _ in 1:num_random_tests
+          x = LO(rand(-10:10, dim(LO)))
+          y = LO(rand(-10:10, dim(LO)))
+          x_ = preimage(iso, x)
+          y_ = preimage(iso, y)
+
+          # mapping non-zero to non-zero
+          @test iszero(x) == iszero(x_)
+          @test iszero(y) == iszero(y_)
+
+          # self-inverse
+          @test iso(x_) == x
+          @test iso(y_) == y
+
+          # f homomorphic
+          @test iso(2 * x_) == 2 * iso(x_)
+          @test iso(x_ + y_) == iso(x_) + iso(y_)
+          @test iso(x_ * y_) == iso(x_) * iso(y_)
+
+          # f^-1 homomorphic
+          @test preimage(iso, 2 * x) == 2 * preimage(iso, x)
+          @test preimage(iso, x + y) == preimage(iso, x) + preimage(iso, y)
+          @test preimage(iso, x * y) == preimage(iso, x) * preimage(iso, y)
+        end
+      end
+    end
+
+    @testset "LinearLieAlgebra" begin
+      lie_algebras = [
+        GAP.Globals.LieAlgebra(
+          RG,
+          GAP.evalstr(
+            "[ [ [ 1, 0 ], [ 0, -1 ] ], [ [ 0, 1 ], [ 0, 0 ] ], [ [ 0, 0 ], [ 1, 0] ] ]"
+          ),
+        ),
+        GAP.Globals.LieAlgebra(
+          RG,
+          GAP.evalstr(
+            "[ [ [ 1, 0 ], [ 0, -1 ] ], [ [ 0, 1 ], [ 0, 0 ] ], [ [ 0, 0 ], [ 1, 0] ] ]"
+          ),
+          GAP.Obj("basis"),
+        ),
+      ]
+
+      @testset for LG in lie_algebras
+        iso = Oscar.iso_gap_oscar(LG)
+        @test domain(iso) == LG
+        LO = codomain(iso)
+        @test LO isa LinearLieAlgebra
+        @test codomain(Oscar.iso_oscar_gap(LO)) === LG
```

Comment:
Perhaps put the tests into a function, then the code duplication can be avoided.